### DISCUSSION TITLE
Render full page in a variable from component php

### DISCUSSION THREAD

**Discussion by xitara** [wintercms/winter#80] [Q&A]
Hi, I want to render a CMS-page in a component with php into a variable. The following code renders the page to a variable but breaks the output in browser (empty body)

```php
$controller = new Controller;
$theme = Theme::getActiveTheme();
$page = Page::load($theme, $config['page']);
var_dump($controller->runPage($page, false));
```

same with

```php
var_dump($controller->runPage($page, false));
```

and

```php
var_dump($controller->renderContent($config['page']));
```

and

```php
Controller::render($config['page']);
```

How can I render a page in php without breaking page-output?

thx4hlp

**Comment by bennothommo** [comment]
@xitara so we can determine if this is the best course of action, what are you trying to achieve by loading a full page to a variable?

**Reply by LukeTowers** [reply]
@bennothommo I was literally just trying to do this a couple weeks ago, I can't really remember why though...

**Reply by bennothommo** [reply]
@LukeTowers if I recall correctly, you were proposing a ChildPage component at one stage (I think this was for the Pages plugin though).

**Reply by LukeTowers** [reply]
Yeah, there was definitely that, I might have been trying to do something with the documentation on the marketplace for this case though.

**Reply by xitara** [reply]
@bennothommo I have a plugin to place boxes/sections on a page and i wand to render a whole page and put it in one of the boxes (see screenshot)

![Bildschirmfoto_2021-04-19_09-42-49](https://user-images.githubusercontent.com/26457767/115199813-dec6ce00-a0f3-11eb-876c-0f1c89d47f85.png)). 

There are some komplex components in the pages an I need the rendered components

I hope you understand what I mean.

**Reply by bennothommo** [reply]
@xitara Just to clarify, when you mean rendering the whole page, do you mean including the layout as well, or just the page itself?

**Reply by xitara** [reply]
sorry for the late reply.
the page without layout is also ok, with layout would be a good option.
including layout give me the possibility to render different pages with a global structure (certenly without <head/> etc.
